### PR TITLE
Add beginnings of mixed code execution support

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -7915,7 +7921,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 		MONO_EMIT_NEW_CHECK_THIS (cfg, arg_ins->dreg);
 	}
 
-	skip_dead_blocks = !dont_verify;
+	skip_dead_blocks = FALSE; //!dont_verify;
```
Comment (kumpera) [inline mono/mini/method-to-ir.c]: Why this?

**Comment by lewurm** [comment]
@monojenkins build interpreter

File: mono/mini/method-to-ir.c (hunk 1 above)
Comment (vargaz) [inline mono/mini/method-to-ir.c]: oversight.

File: mono/mini/mini-generic-sharing.c [hunk 2]
```diff
@@ -1408,6 +1408,116 @@ mini_get_gsharedvt_out_sig_wrapper (MonoMethodSignature *sig)
...
+	WrapperInfo *info;
+	MonoMethodSignature *csig, *gsharedvt_sig;
+	int i, pindex, retval_var = 0;
+	static GHashTable *cache;
+
+	sig = mini_get_underlying_signature (sig);
+
+	if (!cache)
+		cache = g_hash_table_new_full ((GHashFunc)mono_signature_hash, (GEqualFunc)mono_metadata_signature_equal, NULL, NULL);
+	gshared_lock ();
```
Comment (lewurm) [inline mono/mini/mini-generic-sharing.c]: shouldn't we lock before we create a new hashtable?

File: mono/mini/interp/interp.c [hunk 3]
```diff
@@ -1525,84 +1729,211 @@ do_icall (ThreadContext *context, int op, stackval *sp, gpointer ptr)
...
-	if (!method_pointer_hash) {
-		// FIXME: is registering method table as GC root really necessary?
-		// MONO_GC_REGISTER_ROOT_FIXED (method_pointer_hash);
-		method_pointer_hash = g_hash_table_new (NULL, NULL);
-	}
-	addr = g_hash_table_lookup (method_pointer_hash, method);
-	if (addr) {
-		mono_os_mutex_unlock (&create_method_pointer_mutex);
-		return addr;
+	if (method->wrapper_type && method->wrapper_type != MONO_WRAPPER_RUNTIME_INVOKE)
```
Comment (lewurm) [inline mono/mini/interp/interp.c]: any reason why it's limited to `RUNTIME_INVOKE`?  shouldn't we have this exception only for stuff like `WRAPPER_SUBTYPE_INTERP_IN`?

Comment (vargaz) [inline mono/mini/interp/interp.c]: Thats the only one which is tested.
The INTERP_IN stuff doesn't get to this function, its always jitted.

**Comment by lewurm** [comment]
@monojenkins build interpreter

**Pull request merged**